valhalla/midgard/vector2.h

```diff
@@ -14,51 +14,52 @@ namespace midgard {
  * 2D vector class. float x,y components.
  * @author  David W. Nesbitt
  */
-class Vector2 {
+template <typename PrecisionT> class VectorXY {
```

Comment:
instead of renaming this class and then doing `using Vector2 = VectorXY<float>;` below could we change the default template value to be `float`? ie: `template <typename PrecisionT=float> class Vector2 {`

ah crap. just checked this. seems like pre c++17 you still have to at least put the template operator `Vector2<> foo;` for it to compile. in c++17 and above it will automatically deduce the template argument using the default and doing just `Vector2 foo;` will do the right thing.

Reply:
Unfortunately, yep :-(  I've taken a less invasive approach on a new local branch, still trying to untangle this bit, but it has a lot fewer changes.

Reply:
roger that, let me know when i should take a look again!